mono/metadata/marshal-ilgen.c

```diff
@@ -704,6 +706,10 @@ emit_object_to_ptr_conv (MonoMethodBuilder *mb, MonoType *type, MonoMarshalConv
 
 		if (type->type == MONO_TYPE_SZARRAY) {
 			eklass = type->data.klass;
+		} else if (type->type == MONO_TYPE_ARRAY) {
+			eklass = type->data.array->eklass;
+			if (!eklass->blittable)
+				g_assert_not_reached ();
```

Comment:
Is there a test for this ?

Reply:
We have tests in unity for this. I will try to bring them over to your test suite.

Reply:
Please have `g_assert(eklass->attribute)`. It will give more information in case of crash and if we don't have the stacktrace.

Reply:
You mean `g_assert(eklass->blittable)`?

Reply:
Yes, `g_assert(eklass->blittable)`, sorry.